lib/publiccloud/instance.pm

```diff
@@ -278,14 +278,15 @@ Return C<1> true explicit, as stateless and never impact calling code.
 
 sub upload_check_logs_tar {
     my ($self, @files) = @_;
-    my $remote_tar = $autotest::current_test->{name};
-    $remote_tar = "/tmp/" . $remote_tar . "_logs.tar.gz";
-    my $res = $self->ssh_script_output(cmd => 'sudo ls ' . join(' ', @files) . ' 2>/dev/null', proceed_on_failure => 1);
-    return 1 unless ($res);
-
+    my $remote_tar = "/tmp/" . $autotest::current_test->{name} . "_logs.tar.gz";
+    my $cmd = 'sudo ls -x ' . join(' ', @files) . " 2>/dev/null";
+    my $res = $self->ssh_script_output(cmd => $cmd, proceed_on_failure => 1);
+    my @logs = split(" ", $res);
```

Comment:
```suggestion
    my @logs = split("\n", $res);
```
Did your code work? The output of `ls` should print each file on separate **line** or?

Reply:
Yes my cose work well as by VR in header, I also explained it in previous post, I used ls -x that prints files horizontally space separated, so splitted by space it works 🙂